sdk/core/core/inc/az_span.h

```diff
@@ -84,30 +88,37 @@ AZ_NODISCARD AZ_INLINE int32_t az_span_capacity(az_span span) { return span._int
   }
 
 /**
- * @brief Creates a az_span expression. For example,
- * `some_function(AZ_SPAN_FROM_STR("Hello world"));`
- * where
- * `void some_function(const az_span span);`
+ * @brief The AZ_SPAN_FROM_STR macro returns an az_span expression over a literal string. For
+ * example: `some_function(AZ_SPAN_FROM_STR("Hello world"));` where `void some_function(const
+ * az_span span);`
  */
 #define AZ_SPAN_FROM_STR(STRING_LITERAL) (az_span) AZ_SPAN_LITERAL_FROM_STR(STRING_LITERAL)
 
 /**
- * @brief Init span
+ * @brief az_span_init returns a span over a byte buffer.
  *
+ * @param[in] ptr The memory address of the 1st byte in the byte buffer
+ * @param[in] length The number of bytes initialized in the byte buffer
+ * @param[in] capacity The number of total bytes in the byte buffer
+ * @return az_span The az_span "view" over the byte buffer
  */
 AZ_NODISCARD az_span az_span_init(uint8_t* ptr, int32_t length, int32_t capacity);
 
 /**
- * @brief Creates a span from zero terminated string
+ * @brief az_span_from_str returns an az_span from a 0-terminated array of bytes (chars)
  *
+ * @param[in] str The pointer to the 0-terminated array of bytes (chars)
+ * @return az_span An az_span over the byte buffer; length & capacity are set to the string's
+ * length.
  */
 AZ_NODISCARD az_span az_span_from_str(char* str);
 
 /**
- * Creates an az_span literal which can be used to initialize a constant. For example
+ * @brief AZ_SPAN_LITERAL_FROM_BUFFER returns a literal az_span over an unitialized byte buffer. For
+ * example:
  *
- * uint8_t buffer[2 * 1024];
- * const az_span foo = AZ_SPAN_LITERAL_FROM_BUFFER(buffer);
+ * uint8_t buffer[1024];
+ * const az_span buf = AZ_SPAN_LITERAL_FROM_BUFFER(buffer);  // Len=0, Cap=1024
  */
 #define AZ_SPAN_LITERAL_FROM_BUFFER(BYTE_BUFFER) \
```

Comment:
I am having trouble understanding the need for both `AZ_SPAN_LITERAL_FROM_BUFFER` and `AZ_SPAN_FROM_BUFFER`. Why not just have `AZ_SPAN_FROM_BUFFER`?

Reply:
Both required due to a compiler issue. Statically initializing a struct (this is the definition of a 'literal struct') won't compile with the cast. 
Passing an initialized struct (created at runtime; so not a literal struct) won't compile if the ast is not present.

So, we need both macros. I'm unhappy about this too, believe me but we really had no choice.

Reply:
I see. Interesting. So, the difference between the two macros is that explicit cast. I am trying to understand this in the two cases you mentioned (especially with writing code at the call site).

When doing so, I get a compiler error. Any ideas what I am doing wrong?

```c
int Temp(uint8_t buff[])
{
  az_span span = AZ_SPAN_FROM_BUFFER(buff);

  az_span literal = AZ_SPAN_LITERAL_FROM_BUFFER(buff);

  return az_span_length(span) + az_span_length(literal);
}
```

```text
test_span.c(16): warning C6384: Dividing sizeof a pointer by another value. 
test_span.c(18): warning C6384: Dividing sizeof a pointer by another value. 
```